### ISSUE TITLE
[alpha] Cannot use Transition component imported from MUI

### ISSUE THREAD

**Issue by spjy** [iamhosseindhv/notistack#513]
I am trying to pass in a transition into the SnackbarProvider prop `TransitionComponent`. In the latest stable build, 2.0.5, it works if I just pass in MUI's Grow transition component like how the docs currently say.

In 3.0.0-alpha.7, I get an error with the same implementation (on MUI v5.8.4):

```  
import { Grow } from '@mui/material';

<SnackbarProvider
   TransitionComponent={Grow}
>
   <App />
</SnackbarProvider>
```
```
No overload matches this call.
  Overload 1 of 2, '(props: SnackbarProviderProps | Readonly<SnackbarProviderProps>): SnackbarProvider', gave the following error.
    Type 'Element' is not assignable to type 'JSXElementConstructor<TransitionProps & { children: ReactElement<any, any>; }> | undefined'.
  Overload 2 of 2, '(props: SnackbarProviderProps, context: any): SnackbarProvider', gave the following error.
    Type 'Element' is not assignable to type 'JSXElementConstructor<TransitionProps & { children: ReactElement<any, any>; }> | undefined'.
```

The docs still say "If you already have Material-UI as a dependency in your project, you can simply use [MUI transitions](https://mui.com/material-ui/transitions)." If the implementation changed between versions, it would be useful to have an example that reflects it.

**Comment by iamhosseindhv** [comment]
That's a mistake in the documentation. For now if you've upgraded to v3 alpha, you can't use MUI transitions and have to go through the guide [here](https://notistack.com/features/transition) to define custom transitions like Grow.

I'll keep this open to tackle the issue.

**Comment by spjy** [comment]
Are there plans to support MUI transitions like that in v3? That is a lot of needless, custom logic when I could just import the Grow component and pass it in.

**Comment by iamhosseindhv** [comment]
Yup I agree with you @spjy . I'll work on it during the next week or so and get a fix out, so people can easily import MUI transitions, if they already have it installed.

**Comment by iamhosseindhv** [comment]
Hey @spjy , thanks for reporting this bug. A fix has been published as version `3.0.0-alpha.8` (`npm i notistack@alpha`). I've also updated the [relevant page](https://notistack.com/features/transition).

Please let me know if there're any issues.
